### PR TITLE
Don't use cudnnConvolutionBiasActivationForward.

### PR DESCRIPTION
**Pull request by galv**
It supports only the precomputed implicit GEMM implementation of
convolution.

Also fix the bias layout.

### CODE REVIEW

File: src/nnet3/convolution-cudnn.cc [hunk 1]
```diff
@@ -232,35 +231,30 @@ void ConvolutionComputation::InitCudnn() {
...
   CUDNN_SAFE_CALL(
     cudnnSetTensorNdDescriptor(output_desc_, CUDNN_DATA_BASEFLOAT, 4, out_dims,
-			       out_stride));
+                               out_stride));
 
   // We pad the bias with leading dims of 1, since CUDNN's tensors appear to
   // need a dimension of at least 3.
-  int bias_dims[3] = {1, 1, c.num_channels_out};
-  int bias_stride[3] = {c.num_channels_out, c.num_channels_out, 1};
+  int bias_dims[4] = {1, c.num_channels_out, 1, 1};
```
Comment (danpovey) [inline src/nnet3/convolution-cudnn.cc:246]: Can you explain - preferably in a comment - why this specific pattern of dims is required? It's mysterious to me.

Comment (galv) [reply src/nnet3/convolution-cudnn.cc:246]: Done. I can make the bias optional at some point later.

**Comment by galv** [comment]
@danpovey I made the bias optional in my latest commit. The ConvolutionComputation object doesn't know whether the bias is optional in my design. It will simply avoid running any bias-related routines if the bias argument points to null. If you want the ConvolutionComputation object to take a boolean specifying whether or not the bias is used, and check that that matches with whether the bias is null or not, feel free to implement it that way.

**Comment by danpovey** [comment]
Thanks-- merging.  Try to get in the habit of changing the documentation when you make code changes!  It's easier psychologically to do it right away than later.

**Pull request merged**